### DISCUSSION TITLE
Ring doorbell notifications firing, but events not captured by subscription, and videos not recording to file without setting ffmpeg path

### DISCUSSION THREAD

**Discussion by SingleEntity** [dgreif/ring#1467] [Q&A]
Firstly, thank you for the hard work in creating and maintaining this library, its brilliant.

I have two issues unfortunately (13.0.1):

#1 I have two devices, a ring doorbell and a ring camera, the ring camera subscriptions to onMotionDetected work fine, but the one for my doorbell does not.  The code is identical for both cameras.  Why might it be that one device receives motion detection and the other doesn't.  Both are at the same location, both are freshly added to my ring account, with a new token was generated and refresh tokens generated thereafter, yet doorbell ring event and motion events are not firing for the doorbell only.  Any ideas would be welcome!

#2 This took me a while to suss out, but video recordings were not resulting in a file.  The promise was resolving correctly, but no file was being written using the recordToFile method.  Adding the ffmpeg path explicitly solved the issue, but hours of struggling led me to that discovery by accident, and so I thought I'd mention it here in case it helps somebody else since I had no errors, just no file.

**Comment by tsightler** [comment]
Please follow:  https://github.com/dgreif/ring/wiki/Notification-Troubleshooting#steps-to-fix-notifications

This fixes 99.99% of all push notification issues.  Also, it's not 100% clear what you mean by:

> with a new token was generated and refresh tokens generated thereafter

You must always save all updated refresh tokens to disk for future use as they contain the static push notification registration data.  Your case is a little unique in that you are seeing notifications for one camera but not another, it's really difficult to see how this is possible as everything just comes via the same push receiver.  Make sure you have device wide notifications enabled, no restrictions like smart notifications, etc.

**Comment by PeterBunBun** [comment]
Hi @SingleEntity, I'm not sure if this is still an issue to you, but just sharing my 2 cents here. There's an instance for me that I experienced similar behavior as your #1, that's when I turned off the "Smart Notification" only on a Doorbell

Using iPhone Ring app as example:
1. Click into any camera, doorbell...etc
2. The first "Motion Detection" button appears here is for the current device, namely the iPhone, so this one doesn't matter
3.  Enter "Motion Settings"
4. Enter "Smart Alerts". The "bell" icon you see here could turned on/off notifications for all listening device on this camera, doorbell (cutting it off from the source)

With this turned off (and the other camera might be in on status), my setup exhibit similar behavior as your #1
